### PR TITLE
Set up Prettier

### PR DESCRIPTION
**Pull request by mitas1**
~Improves ESLint configuration (most of these are just stylistic preferences) and~ sets up Prettier.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,10 +31,14 @@
...
     "eslint-config-airbnb": "^19.0.4",
     "eslint-config-airbnb-typescript": "^17.0.0",
+    "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-prefer-arrow": "^1.2.3",
+    "eslint-plugin-prettier": "^4.2.1",
     "eslint-plugin-rulesdir": "^0.2.1",
     "eslint-plugin-sonarjs": "^0.16.0",
     "eslint-webpack-plugin": "^3.2.0",
     "gatsby-plugin-eslint": "^4.0.2",
+    "prettier": "2.7.1",
```
Comment (lubo) [inline package.json:41]: I see no script, how do you use it?

File: .eslintrc.js [hunk 2]
```diff
@@ -60,6 +59,19 @@ module.exports = {
...
+          'error',
+          {
+            namedComponents: 'arrow-function',
+            unnamedComponents: 'arrow-function',
+          },
+        ],
+        'arrow-body-style': 'error',
+        'react/require-default-props': 'off',
+        'react/prop-types': 'off',
+      },
```
Comment (lubo) [inline .eslintrc.js:74]: 1. Why did you add these rules to overrides instead of applying them to all the files?
2. Can you explain why do you prefer using these rules? Why should we use the bottom two when they forgo type safety?

Comment (mitas1) [reply .eslintrc.js:74]: 1. That's a good question. I've tried, but some of the rules are later overridden by other plugins. I can put the rules into a variable and put it in both places. Or do you have a better suggestion?
2. The bottom two are not useful with Typescript.

File: package.json (hunk 1 above)
Comment (mitas1) [reply package.json:41]: I'm using it in my VS code editor: https://github.com/prettier/prettier-vscode, whenever I need to format a code. I understand your point, it is possible to keep it away from our repo I don't have a strong opinion on it, but it is recommended by prettier docs https://prettier.io/docs/en/install.html to have it installed locally. I can add the following script with `.prettierignore` to our repo:
```sh
prettier --write .
```
Let me know

Comment (lubo) [reply package.json:41]: I see. Can you add something like `"format": "prettier --write ."` to `scripts` then?

File: .eslintrc.js (hunk 2 above)
Comment (lubo) [reply .eslintrc.js:74]: 1. Yes, let's create `commonRules` and do the some thing as with `commonPresets`.
2. OK. But what about the arrow functions? Why do you prefer them?

File: .eslintrc.js [hunk 3]
```diff
@@ -8,18 +8,20 @@ const rulesDirPlugin = require('eslint-plugin-rulesdir');
...
+rulesDirPlugin.RULES_DIR = path.resolve(
+  path.join(
+    __dirname,
+    'node_modules',
+    'gatsby',
+    'dist',
+    'utils',
+    'eslint-rules',
+  ),
+);
```
Comment (lubo) [inline .eslintrc.js:20]: If we don't prefer the former, then there should probably be a linter rule for that.

File: .eslintrc.js (hunk 2 above)
Comment (lubo) [reply .eslintrc.js:74]: I've created #9 for point 1, as it's currently broken in `main` too. Merge it into this branch after it's merged into `main`.

Comment (mitas1) [reply .eslintrc.js:74]: > OK. But what about the arrow functions? Why do you prefer them?

It is a stylistic preference. I'm used to using arrow functions.

File: .eslintrc.js (hunk 3 above)
Comment (mitas1) [reply .eslintrc.js:20]: Can you elaborate a bit more? I'm not sure what are you referring to.

File: package.json (hunk 1 above)
Comment (mitas1) [reply package.json:42]: Sure, I've added it to the `scripts`.

File: .eslintrc.js (hunk 3 above)
Comment (lubo) [reply .eslintrc.js:20]: If this code needs reformatting, then there should be a linter rule telling us that.

Comment (lubo) [reply .eslintrc.js:20]: Previously, I didn't know eslint-plugin-prettier would make ESLint tell us if somethings wrong.

File: .prettierignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+.cache/
+public
+src/gatsby-types.d.ts
```
[the remark this answers is not in the document]
Comment (lubo) [reply .prettierignore]: Done that in 620c2aff18d15d0593176cb79176b138d7d658db.

File: tsconfig.json [hunk 5]
```diff
@@ -11,9 +11,12 @@
...
     /* Language and Environment */
-    "target": "esnext",                                  /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */
-    "lib": ["dom", "esnext"],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
-    "jsx": "react",                                /* Specify what JSX code is generated. */
+    "target": "esnext" /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+    "lib": [
+      "dom",
+      "esnext"
+    ] /* Specify a set of bundled library declaration files that describe the target runtime environment. */,
+    "jsx": "react" /* Specify what JSX code is generated. */,
```
[the remark this answers is not in the document]
Comment (lubo) [reply tsconfig.json:19]: It's a would-be-nice, let's not waste time with that now.

**Review by lubo — APPROVED** [review]
Let's just use Prettier.
